### ISSUE TITLE
Calculate errors recursively

### ISSUE THREAD

**Issue by kevinsuedmersen** [kevinsuedmersen/library-neuralnetworksfromscratch#1]
Calculate the errors recursively from layer L-1 to layer 1

Tasks:
- [x] Compute Jacobians of the activations in layer l w.r.t. the dendritic potentials in layer l for the following activation functions:
    - [x] Softmax
    - [x] linear
    - [x] ReLU
    - [x] Sigmoid
    - [x] tanh
- [x] Implement the `backward` method of the `DenseLayer` class using the Jacobians, weight matrices and errors in layer l+1
- [x] Test the `backward` method of the `DenseLayer` class
- [x] Move code from `interface.py` into `__init__.py`

Tests:
- [x] Make sure that shapes of backpropagated errors make sense
